### PR TITLE
EASY-1212

### PR DESCRIPTION
**Pull request by janvanmansum**
Now calling the (refactored) DepositHandler.createDepositReceipt function from the Edit-IRI as well

fixes EASY-1212

#### When applied it will
* Make sure the client gets a full deposit receipt whevener getting the Edit-IRI

#### Where should the reviewer @DANS-KNAW/easy start?
https://github.com/DANS-KNAW/easy-sword2/compare/master...janvanmansum:EASY-1212?expand=1#diff-3da2c290b7fe234963377b62a410aed1

#### How should this be manually tested?
See JIRA issue.

### CODE REVIEW

File: src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala [hunk 1]
```diff
@@ -437,7 +439,7 @@ object DepositHandler {
       .flatten
   }
 
-  def createDepositReceipt(deposit: Deposit, settings: Settings, id: String): DepositReceipt = {
+  def createDepositReceipt(settings: Settings, id: String): DepositReceipt = {
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala:442]: I recently found a new trick in Scala that would probably nicely go with this method:

```
def createDepositReceipt(settings: Settings, id: String): DepositReceipt = {
  new DepositReceipt {
    val editIRI = new IRI(settings.serviceBaseUrl + "container/" + id)
    val editMediaIri = new IRI(settings.serviceBaseUrl + "media/" + id)
    val stateIri = settings.serviceBaseUrl + "statement/" + id
    setEditIRI(editIRI)
    setLocation(editIRI)
    setEditMediaIRI(editMediaIRI)
    setSwordEditIRI(editIRI)
    setAtomStatementURI(stateIri)
    setPackaging(Collections.singletonList("http://purl.org/net/sword/package/BagIt"))
    setTreatment("[1] unpacking [2] verifying integrity [3] storing persistently")
  }
}
```

Comment (rvanheest) [reply src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala:442]: btw, is the difference between `IRI` and `Iri` on purpose?

File: src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala [hunk 2]
```diff
@@ -449,11 +451,9 @@ object DepositHandler {
     dr.setAtomStatementURI(stateIri)
     dr.setPackaging(Collections.singletonList("http://purl.org/net/sword/package/BagIt"))
     dr.setTreatment("[1] unpacking [2] verifying integrity [3] storing persistently")
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala:453]: Is this just a String that is printed/returned somewhere, or are the `[1]... [2]... [3]...` special formatting? It's not part of this PR, but I'm just curious about it.

File: src/main/scala/nl.knaw.dans.easy.sword2/ContainerManagerImpl.scala [hunk 3]
```diff
@@ -36,19 +36,10 @@ class ContainerManagerImpl extends ContainerManager {
     SwordID.extract(editIRI) match {
       case Success(id) =>
         val dir: File = new File(settings.depositRootDir, id)
-        if (!dir.exists) {
-          throw new SwordError(404)
-        }
-      case _ => throw new SwordError(404)
+        if (dir.exists) DepositHandler.createDepositReceipt(settings, id)
```
Comment (rvanheest) [inline src/main/scala/nl.knaw.dans.easy.sword2/ContainerManagerImpl.scala:39]: don't you need the `verboseDescription` in this call to the receipt?

**Review by rvanheest — COMMENTED** [review]
Nice work!

File: src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala (hunk 2 above)
Comment (janvanmansum) [reply src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala:453]: It is just a string.

File: src/main/scala/nl.knaw.dans.easy.sword2/ContainerManagerImpl.scala (hunk 3 above)
Comment (janvanmansum) [reply src/main/scala/nl.knaw.dans.easy.sword2/ContainerManagerImpl.scala:39]: We don't have the MD5 that is displayed in the `verboseDescription` here.
